diff --git a/mozharness/mozilla/building/buildb2gbase.py b/mozharness/mozilla/building/buildb2gbase.py
--- a/mozharness/mozilla/building/buildb2gbase.py
+++ b/mozharness/mozilla/building/buildb2gbase.py
@@ -204,9 +204,9 @@ class B2GBuildBaseScript(BuildbotMixin,

         # Set up mock immediately so any later run_command_m doesn't end up
         # calling setup_mock with the wrong config.
-        self.setup_mock(gecko_config['mock_target'],
-                        gecko_config['mock_packages'],
-                        gecko_config.get('mock_files'))
+        #self.setup_mock(gecko_config['mock_target'],
+        #                gecko_config['mock_packages'],
+        #                gecko_config.get('mock_files'))

         return gecko_config